### PR TITLE
resolves #105 add api to delete an intent associated to an agent

### LINKED ISSUE #105: Add api to delete an intent associated to an agent
Delete an intent associated to the agent 
Should also delete the training data and responses associated to the intent

### PR DESCRIPTION
**Pull request by elisaMy**

### CODE REVIEW

File: api/agents_service.py [hunk 1]
```diff
@@ -228,6 +228,12 @@ def delete_agent(self, agent_name):
         
         self.delete_agent_from_memory(agent_name)
 
+    def delete_agent_intent(self, agent_name, intent_name):
+        """ Delete an intent associated to an agent """
```
Comment (omar-mellouli) [inline api/agents_service.py:232]: Remove useless comment

File: api/persistence/agents_repository.py [hunk 2]
```diff
@@ -44,6 +44,13 @@ def update_agent_intents(self, agent_name, intents):
             {"name": agent_name}, 
             { "$addToSet": { "intents": {"$each":intents} } }
         )
+
+    def delete_agent_intent(self, agent_name, intent):
+        """Delete an agent intent"""
```
Comment (omar-mellouli) [inline api/persistence/agents_repository.py:49]: Remove useless comment

**Review by omar-mellouli — CHANGES_REQUESTED** [review]

File: api/persistence/training_data_repository.py [hunk 3]
```diff
@@ -63,3 +63,6 @@ def get_intent_by_text(self, text, agent_name):
     
     def delete_all_training_data(self, agent_name):
         self.training_data_collection.delete_many({"agent_name": agent_name})
+
+    def delete_intent_training_data(self, agent_name, intent_name):
```
Comment (omar-mellouli) [inline api/persistence/training_data_repository.py:67]: rename:
delete_agent_training_data_by_intent

File: api/persistence/responses_repository.py [hunk 4]
```diff
@@ -25,6 +25,9 @@ def insert_response(self, agent_name, response):
     def delete_all_agent_responses(self, agent_name):
         self.responses_collection.delete_many({"agent_name": agent_name})
 
+    def delete_intent_responses(self, agent_name, intent_name):
```
Comment (omar-mellouli) [inline api/persistence/responses_repository.py:28]: rename:
delete_agent_responses_by_intent

File: api/agents_service.py (hunk 1 above)
Comment (elisaMy) [reply api/agents_service.py]: Done

File: api/persistence/agents_repository.py (hunk 2 above)
Comment (elisaMy) [reply api/persistence/agents_repository.py]: Done

File: api/persistence/training_data_repository.py (hunk 3 above)
Comment (elisaMy) [reply api/persistence/training_data_repository.py]: Done

File: api/persistence/responses_repository.py (hunk 4 above)
Comment (elisaMy) [reply api/persistence/responses_repository.py]: Done

**Review by omar-mellouli — APPROVED** [review]

**Pull request merged**